### PR TITLE
docs: add landing page for rules

### LINKED ISSUE #4121: Create a landing page for Rules
## Suggested Changes

The _Rules_ link in https://typescript-eslint.io's header links to an arbitrary rule file. It should have its own landing page.

Should also include an auto-generated table.

### PR DESCRIPTION
**Pull request by armano2**
## Overview
- basic root `/rules/` page for rules
  - navigation url has been updated
- deprecated / removed rules will be present in different subcategory
- Minor fixes to plugin docs

Fixes #4121

### CODE REVIEW

**Review by bradzacher — COMMENTED** [review]
LGTM!



> deprecated / removed rules will be present different subcategory

We should add a "note" to the `no-implicit-any-catch` docs that mentions why it was deprecated. Something like this will do

> This rule has been deprecated because TypeScript 4.4 introduced the [`useUnknownInCatchVariables`](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables) compiler option which automatically types all catch variables as `unknown` - meaning a lint rule for manual `unknown` annotations is no longer necessary.

File: packages/eslint-plugin/docs/rules/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: Overview
+sidebar_label: Overview
+pagination_next: null
+pagination_prev: null
+slug: /
+---
+
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/README.md:8]: What do you think about inserting a blurb here about the two main rule config lists? Vaguely:

```suggestion

`@typescript-eslint/eslint-plugin` comes with two rulesets you can extend from to pull in the recommended starting rules:

* `'plugin:@typescript-eslint/recommended'`: unopinionated rules for code correctness you can drop in without additional configuration. See [Linting](../../linting/README.md) for more details.
* `'plugin:@typescript-eslint/recommended-requiring-type-checking'` additional recommended rules that require type information. See [Linting](../../linting/TYPED_LINTING.md) for more details.

```

**Review by JoshuaKGoldberg — DISMISSED** [review]
Awesome! 🔥

File: packages/eslint-plugin/docs/rules/README.md (hunk 1 above)
Comment (armano2) [reply packages/eslint-plugin/docs/rules/README.md:8]: for now, i added it as kinda placeholder, i do agree that we should add more description and maybe reorganize this file

**Comment by armano2** [comment]
> We should add a "note" to the `no-implicit-any-catch` docs that mentions why it was deprecated. Something like this will do
> 
> > This rule has been deprecated because TypeScript 4.4 introduced the [`useUnknownInCatchVariables`](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables) compiler option which automatically types all catch variables as `unknown` - meaning a lint rule for manual `unknown` annotations is no longer necessary.

i agree with adding deprecation notice, but that change is unrelated to what this PR aims to do

**Review by bradzacher — APPROVED** [review]

**Pull request merged**